velox/exec/WindowFunction.h

```diff
@@ -15,12 +15,22 @@
  */
 #pragma once
 
+#include "velox/core/Expressions.h"
 #include "velox/exec/WindowPartition.h"
 #include "velox/expression/FunctionSignature.h"
 #include "velox/vector/BaseVector.h"
 
 namespace facebook::velox::exec {
 
+// Represents arguments for window functions. Stores argument type,
+// the ConstantTypedExpr(if it is one) or the column index in the input row
+// (if the argument is a FieldAccessTypedExpr).
+struct WindowFunctionArg {
+  const TypePtr type;
+  const core::ConstantTypedExpr* constantValue;
+  const column_index_t argIndex;
```

Comment:
Maybe std::optional? Constant inputs would have that as std::nullopt.

nit: perhaps, drop 'arg' as it is already part of the struct name